test/shared_utilities/space_varying_inputs.jl

```diff
@@ -0,0 +1,75 @@
+using Test
+import ClimaLand
+using ClimaLand.SpaceVaryingInputs: SpaceVaryingInput
+using ClimaLand: Domains
+
+using ClimaCore: Fields
+using ClimaComms
+
+device = ClimaComms.device()
+context = ClimaComms.SingletonCommsContext(device)
+
+@testset "SpaceVaryingInput" begin
+    FT = Float32
+    zmin = FT(-1.0)
+    zmax = FT(0.0)
+    xlim = FT.((0.0, 10.0))
+    ylim = FT.((0.0, 1.0))
+    zlim = FT.((zmin, zmax))
+    nelements = (1, 1, 10)
+    radius = FT(100)
+    depth = FT(30)
+    n_elements_sphere = (6, 20)
+    npoly_sphere = 3
+    # Spherical Shell
+    shell = Domains.SphericalShell(;
+        radius = radius,
+        depth = depth,
+        nelements = n_elements_sphere,
+        npolynomial = npoly_sphere,
+    )
+
+    box = Domains.HybridBox(;
+        xlim = xlim,
+        ylim = ylim,
+        zlim = zlim,
+        nelements = nelements,
+        npolynomial = 0,
+    )
+    column = Domains.Column(; zlim = zlim, nelements = nelements[3])
+
+    domains = [shell, box, column]
+    analytic_func = (coords) -> 2.0
```

Comment:
we could define this as `analytic_func = (lat, lon) -> 2.0` to make sure that the coords are of the right form for each domain, but like this is also fine :)

Reply:
I wanted to keep this general, because the function gets the coordinate field (which has the values stored as coords.lat, coords.lon, or coords.lat, coords.lon, coords.z, or just coords.z, etc). then we only need one method that expects f(coords) instead of one that unpacks them correctly for each space (and the order of arguments is not fixed).